nima_core/cache.py

```
@@ -0,0 +1,251 @@
+    """
+
+    def __init__(self):
+        self._embeddings: Dict[int, np.ndarray] = {}
+        self._matrix: Optional[np.ndarray] = None
+        self._ids: List[int] = []
+
+    def load(self, memory_ids: List[int], embeddings: List[List[float]]):
+        """Load memories into index."""
+        self._embeddings = {
+            mid: np.array(emb, dtype=np.float32)
+            for mid, emb in zip(memory_ids, embeddings)
```

Ruff (0.15.7)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)